core/pom.xml

```diff
@@ -67,6 +67,12 @@
             <artifactId>expiringmap</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>org.immutables</groupId>
+            <artifactId>value</artifactId>
+            <scope>provided</scope>
```

Comment:
This should probably be `<optional>true</optional>`. While they seem to do the same thing, `provided` scope is typically used for things that are required and will be provided at runtime by the container, whereas optional dependencies are, well, optional. From the [Maven documentation](https://maven.apache.org/guides/introduction/introduction-to-optional-and-excludes-dependencies.html): 

> Optional dependencies are used when it's not really possible (for whatever reason) to split a project up into sub-modules.

The above explanation seems to fit this use case. We can think of this Immutables support as a logically separate module.

Reply:
Oh, yes, isn't it a separate module? I would prefer if it was, with an `ImmutablesPlugin`.

Reply:
It's debatable though, because this whole feature is built without ever referring to immutables as external code of any kind.

Reply:
Immutables should be in test scope, as there is no compile-time dependency at all in the main source tree.

Reply:
Unfortunately moving it to `<scope>test</scope>` breaks Eclipse's annotation processor, as it doesn't expect to find test-only processors.  So I've made it `<scope>provided</scope><optional>true</optional>`, hopefully that works?